### PR TITLE
Userguide and tutorial JAX transformation of Pulse Scheduler

### PR DESCRIPTION
**Pull request by to24toro**
### Summary

This tutorial and user guide are related to https://github.com/Qiskit/qiskit-dynamics/pull/149.
In the user guide, 
- show some pulses using ScalableSymbolicPulse.
- jax-jit these pulses.

In the tutorial,
- show the optimization of a single X gate defined by pulse schedulers.
The tutorial is much overlapping with the configuration of https://qiskit.org/documentation/dynamics/tutorials/optimizing_pulse_sequence.html .
Whether or not merging this PR depends on the discussion.

### Details and comments
Currently, a new class ScalableSymbolicPulse which is a subclass of SymbolicPuse and assumed to hava a scalable form was introduced. So I use this class in this PR. However, I use the SymbolicPulse in https://github.com/Qiskit/qiskit-dynamics/pull/149. I would like to discuss which we should use.

### CODE REVIEW

**Review by DanPuzzuoli — CHANGES_REQUESTED** [review]
I've given direct comments on the user guide. I think it is good, but just needs to be a bit simplified.

For the tutorial, I think it can be much more simplified. I don't think it is necessary to have separate 2-A, ..., 6-A and 2-B, ..., 6-B sections. I was thinking the original 2-6 steps could be kept as they are, and a single additional section numbered 7 (with subsections) could be added for doing the `ScalableSymbolicPulse` optimization, using the same model and fidelity function as sections 2-6. I realize that you are wanting to use DRAG, the relevance of which requires using a transmon model, but for the simplicity of the presentation I think it makes sense to just use the qubit model of section 2.

File: docs/userguide/how_to_use_pulse_schedule_for_jax_jit.rst [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+.. warning::
+    This is an advanced topic --- Some extend of knowledge about ``Qiskit-pulse`` is required.
+    See `Qiskit pulse API Reference <https://qiskit.org/documentation/apidoc/pulse.html>`__ for the detailed  API information.
+    Also, see `Qiskit tutorial <https://qiskit.org/documentation/tutorials/circuits_advanced/06_building_pulse_schedules.html>`__ 
+    for a tutorial about a pulse scheduler.
+
+.. note::
+    At present, only the ``ScalableSymbolicPulse`` class is supported by JAX.
+    We cannot use any subclasses such as ``Gaussian`` and ``GaussianSquare``
+    for JAX.
```
[the remark this answers is not in the document]
Comment (to24toro) [reply docs/userguide/how_to_use_pulse_schedule_for_jax_jit.rst:23]: make senses.

File: docs/userguide/how_to_use_pulse_schedule_for_jax_jit.rst [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+    # tell JAX we are using CPU
+    jax.config.update('jax_platform_name', 'cpu')
+
+    # import Array and set default backend
+    from qiskit_dynamics.array import Array
+    Array.set_default_backend('jax')
+
+
+2. How to generate a variety of pulses by ``ScalableSymbolicPulse``
```
[the remark this answers is not in the document]
Comment (to24toro) [reply docs/userguide/how_to_use_pulse_schedule_for_jax_jit.rst:53]: Certainly, it may be too much to users to show 3 patterns specially.

File: docs/userguide/how_to_use_pulse_schedule_for_jax_jit.rst [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            valid_amp_conditions=sym.And(sym.Abs(_amp) <= 1.0, sym.Abs(_beta) < _sigma),
+        )
+
+    drag_pulse.draw()
+
+
+4. How to execute jit compilation using ``ScalableSymbolicPulse``
+-----------------------------------------------------------------
+
+Using a Gaussian pulse as an example, jit compilation is executed as follows.
```
[the remark this answers is not in the document]
Comment (to24toro) [reply docs/userguide/how_to_use_pulse_schedule_for_jax_jit.rst]: Thanks for adding the sentence. Modified at https://github.com/Qiskit/qiskit-dynamics/pull/194/commits/b7ef506fb906a3b9763c99c0b40158f4fafd95ad.

**Review by DanPuzzuoli — CHANGES_REQUESTED** [review]
This current iteration is looking good. I've suggestions for edits to the text.

Aside from these suggestions, one thing you should do is add direct sphinx references to classes/functions in Terra. E.g. in the [`DynamicsBackend` docs](https://github.com/Qiskit/qiskit-dynamics/blob/c1f160a9e6dfbbb5937ba393cd0218c0bd71f962/qiskit_dynamics/backend/dynamics_backend.py#L65), there are references to objects in terra, such as:
```
:class:`~qiskit.providers.backend.BackendV2`
```

(As an aside, in case you haven't seen this before, [this line in docs/conf.py](https://github.com/Qiskit/qiskit-dynamics/blob/c1f160a9e6dfbbb5937ba393cd0218c0bd71f962/docs/conf.py#L74) sets up the `intersphinx` package for linking documentation to other packages. In this case it is only `qiskit`, but `qiskit_experiments` is also being added in #193 .)

So, e.g., all of the references to ``ScalableSymbolicPulse`` can be turned into links to the API documentation for the class. I think you should even be able to link to the qiskit pulse module using
```
:mod:`~qiskit.pulse`
```
(not sure if this is correct syntax, but something like this should work).

File: docs/tutorials/optimizing_pulse_sequence.rst [hunk 4]
```diff
@@ -254,3 +256,127 @@ approximation analysis.
...
+    opt_results = minimize(fun=jit_grad_obj, x0=initial_params, jac=True, method='BFGS')
+
+    print(opt_results.message)
+    print(f"Optimized Sigma is {opt_results.x[0]} and Width is {opt_results.x[1]}")
+    print('Number of function evaluations: ' + str(opt_results.nfev))
+    print('Function value: ' + str(opt_results.fun))
+
+
+
+We can draw the optimized pulse, whose parameters are retrieved by :math:`opt_results.x`.
```
Comment (DanPuzzuoli) [inline docs/tutorials/optimizing_pulse_sequence.rst:378]:
```suggestion
We can draw the optimized pulse, whose parameters are retrieved by ``opt_results.x``.
```

**Review by DanPuzzuoli — CHANGES_REQUESTED** [review]
Some final small changes. After these I will approve.

File: docs/tutorials/optimizing_pulse_sequence.rst (hunk 4 above)
Comment (to24toro) [reply docs/tutorials/optimizing_pulse_sequence.rst]: Fix at https://github.com/Qiskit/qiskit-dynamics/pull/194/commits/83684a225937efdbca0a252d3a1bf7e08cf0bab8.

File: docs/tutorials/optimizing_pulse_sequence.rst [hunk 5]
```diff
@@ -254,3 +256,127 @@ approximation analysis.
...
     opt_signal.samples.sum()
+
+
+7.  Repeat the :math:`X`-gate optimization, alternatively using pulse schedules to specify the control sequence.
+----------------------------------------------------------------------------------------------------------------
+
+Here, we perform the optimization again, however now we specify the parameterized control sequence to optimize as a pulse schedule.
+
+We construct a Gaussian square pulse as a :class:`~qiskit.pulse.library.ScalableSymbolicPulse` instance, parameterized by ``sigma`` and ``width``.
+Although qiskit pulse provides a :class:`~qiskit.pulse.library.GaussianSquare`, this class is not JAX compatible.
```
[the remark this answers is not in the document]
Comment (to24toro) [reply docs/tutorials/optimizing_pulse_sequence.rst:267]: Fix at https://github.com/Qiskit/qiskit-dynamics/pull/194/commits/83684a225937efdbca0a252d3a1bf7e08cf0bab8.

File: docs/tutorials/optimizing_pulse_sequence.rst [hunk 6]
```diff
@@ -254,3 +256,127 @@ approximation analysis.
...
+            t_span=[0, instance.duration * dt],
+            signals=[signal],
+            method='jax_odeint',
+            atol=1e-8,
+            rtol=1e-8
+        )
+        return 1. - fidelity(Array(result[0].y[-1])).data
+
+
+We set the initial values of `sigma` and `width` for the optimization as :math:`initial_params = np.array([10, 10,])`.
```
[the remark this answers is not in the document]
Comment (to24toro) [reply docs/tutorials/optimizing_pulse_sequence.rst]: Fix at https://github.com/Qiskit/qiskit-dynamics/pull/194/commits/83684a225937efdbca0a252d3a1bf7e08cf0bab8.

File: docs/tutorials/optimizing_pulse_sequence.rst [hunk 7]
```diff
@@ -254,3 +256,127 @@ approximation analysis.
...
+
+.. jupyter-execute::
+
+    initial_params = np.array([10, 10,])
```
[the remark this answers is not in the document]
Comment (to24toro) [reply docs/tutorials/optimizing_pulse_sequence.rst]: Fix at https://github.com/Qiskit/qiskit-dynamics/pull/194/commits/83684a225937efdbca0a252d3a1bf7e08cf0bab8.

**Review by DanPuzzuoli — APPROVED** [review]
